### PR TITLE
Feat/api metrics issue 13

### LINKED ISSUE #13: API Metrics
Track things like 
- Users usage (by endpoint, overall, etc)
- User approximate geolocation/region
- Anything else is a bonus

**ALL METRICS SHOULD ONLY BE ACCESSED BY ADMIN OR THE USER THAT THE DATA CORRESPONDS TO**

### PR DESCRIPTION
**Pull request by satyap54**
# Pull Request

## Description

With this update, following end-points have been added:

- View history of end-points accessed by a user
- Find out latest location of user
- View usage of each end-point

We now store metrics such as geolocation and end-points accessed by a user in our db.

## Related Issue

Closes #13 

## 6.0.1

Did you remember to update the version number in the package.json file?

- [x] For a major restructuring, or large changes please bump the the major version number. (ex: 1.x.x)
- [ ] For a minor change, bump the minor version number. (ex: x.1.x)
- [ ] For a bug fix, bump the patch version number. (ex: x.x.1)

## Documentation

For _any_ version updates, please verify if the [documentation page]() needs an update. If it does, please [uptade the in code docs]() to ensure it is not forgotten.

- [x] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/routes/api.ts [hunk 1]
```diff
@@ -15,5 +16,6 @@ router.use("/coinflip", coinflipRoutes);
 router.use("/diceroll", dicerollRoutes);
 router.use("/qotd", qotdRoutes);
 router.use("/scam", scamRoutes);
+router.use("/metrics", metricsRoutes);
```
Comment (j-dogcoder) [inline src/routes/api.ts:19]: Please Document Metrics Routes

**Review by j-dogcoder — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ void (async () => {
     console.log(colors.red(`MongoDB Error: \n${err}`));
   });
 
-  const application = await app.listen(PORT, () => {
+  const application = app.listen(PORT, () => {
```
[the remark this answers is not in the document]
Comment (satyap54) [reply src/index.ts:33]: But, this doesn't seem to be an async operation. Do you still need it?

File: src/routes/api.ts (hunk 1 above)
Comment (satyap54) [reply src/routes/api.ts:19]: Added

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (j-dogcoder) [reply src/index.ts:33]: yes

[the remark this answers is not in the document]
Comment (satyap54) [reply src/index.ts]: ok, added

**Comment by satyap54** [comment]
Hi @j-dogcoder , any update on this PR?

**Comment by satyap54** [comment]
No, you can merge it if testing goes well.

**Review by j-dogcoder — DISMISSED** [review]

**Pull request merged**